### PR TITLE
add test case for codeblock

### PR DESCRIPTION
**Pull request by zora-wang**
@monikaprabhu 
@riteshchandawar 
This test case is to test CodeBlockReference.dyn in Colins workshop

### CODE REVIEW

**Comment by zora-wang** [comment]
cbn

**Comment by monikaprabhu** [comment]
@zora-wang - please Run the entire WorkflowTest.dll and check if all test pass.

**Comment by zora-wang** [comment]
workflow

**Comment by zora-wang** [comment]
pass image
@monikaprabhu @riteshchandawar

File: test/Libraries/WorkflowTests/ComplexTests.cs [hunk 1]
```diff
@@ -56,5 +64,185 @@ public void IndexOutsideBounds_3399()
...
+            //check preview value of code block node, which contains a list
+            var listCBN = "4c049b29-2792-4a66-ab89-d8fac3161e7d";
+            AssertPreviewCount(listCBN, 4);
+
+            //check preview value of CBN, which is a list and not null
+            var notNullCBN = "d71e5f04-9928-437a-863c-5bf34666370b";
+            AssertPreviewCount(notNullCBN, 5);
+            for (int i = 0; i < 5; i++)
+            {
+                var element = GetPreviewValueAtIndex(notNullCBN, i) as string;
```
Comment (riteshchandawar) [inline test/Libraries/WorkflowTests/ComplexTests.cs]: as string, does this really work?

**Comment by riteshchandawar** [comment]
LGTM! from my side.

@monikaprabhu Please review this PR, once done, assign this to @ke-yu, he will help in merging this PR.

File: test/Libraries/WorkflowTests/ComplexTests.cs (hunk 1 above)
Comment (zora-wang) [inline test/Libraries/WorkflowTests/ComplexTests.cs]: The test case can pass

**Comment by monikaprabhu** [comment]
@zora-wang , Please add verification Action on Point and User defined function call in CodeBlockReference.dyn

**Comment by monikaprabhu** [comment]
@zora-wang , there is merge conflict in this PR. can you please resolve and send the PR

**Comment by zora-wang** [comment]
okay

**Comment by zora-wang** [comment]
passimage3

**Comment by monikaprabhu** [comment]
@ke-yu - Please review  Commit Update file for conflict - commit "570a9b8".

**Comment by zora-wang** [comment]
passed test case
@monikaprabhu 
@ke-yu , Please help to review the Dynamo unit test cases.

**Comment by monikaprabhu** [comment]
@zora-wang - Please include additional test for transpose and check that the points are transposed correctly.

**Comment by zora-wang** [comment]
@monikaprabhu 
1.the bug has been recorded in http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7391

2. test cases have been updated

**Comment by ke-yu** [comment]
LGTM.

**Pull request merged**